### PR TITLE
Server time zone

### PR DESCRIPTION
**Pull request by yurabakhtin**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [ ] All tests are passing
- [ ] Changelog was modified

**Other information:**
https://github.com/humhub/humhub-internal/issues/386

### CODE REVIEW

File: protected/humhub/libs/DynamicConfig.php [hunk 1]
```diff
@@ -98,10 +98,10 @@ public static function rewrite()
             $config['language'] = Yii::$app->language;
         }
 
-        $defaultTimeZone = Yii::$app->settings->get('defaultTimeZone');
-        if (!empty($defaultTimeZone)) {
-            $config['timeZone'] = $defaultTimeZone;
-            $config['components']['formatter']['defaultTimeZone'] = $defaultTimeZone;
+        $serverTimeZone = Yii::$app->settings->get('serverTimeZone');
```
Comment (luke-) [inline protected/humhub/libs/DynamicConfig.php:101]: @yurabakhtin The long-term goal is to eliminate DynamicConfig. 

It probably makes sense to remove the time zone change here as part of this refactoring.

Perhaps it makes sense to develop a “LocalizationService” that handles both time zones and language settings centrally and cleanly. Database + app.

**Review by luke- — CHANGES_REQUESTED** [review]

File: protected/humhub/libs/DynamicConfig.php (hunk 1 above)
Comment (yurabakhtin) [reply protected/humhub/libs/DynamicConfig.php]: @luke- Ok, I have removed the options `timeZone` and `components / formatter / defaultTimeZone` from the dynamic config file.

File: protected/humhub/modules/installer/forms/LocalisationForm.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        ];
+    }
+
+    public function save(): bool
+    {
+        if (!$this->validate()) {
+            return false;
+        }
+
+        Yii::$app->db;
```
Comment (luke-) [inline protected/humhub/modules/installer/forms/LocalisationForm.php:66]: Is this required?

File: protected/humhub/modules/user/models/User.php [hunk 3]
```diff
@@ -276,6 +278,10 @@ public function __get($name)
             return $profile;
         }
 
+        if ($name === 'time_zone' && empty(parent::__get($name))) {
```
Comment (luke-) [inline protected/humhub/modules/user/models/User.php:281]: Maybe a comment would be nice. "e.g. fall back to default".

**Review by luke- — CHANGES_REQUESTED** [review]

File: protected/humhub/libs/DynamicConfig.php (hunk 1 above)
Comment (luke-) [reply protected/humhub/libs/DynamicConfig.php]: @yurabakhtin Can you please also "unset" these options, on the DynamicConfig save.

File: protected/humhub/components/ApplicationTrait.php [hunk 4]
```diff
@@ -49,6 +48,13 @@ public function __construct($config = [])
         $config = $this->removeLegacyConfigSettings($config);
 
         parent::__construct($config);
+
+        if ($this->isDatabaseInstalled(true)) {
```
[the remark this answers is not in the document]
Comment (yurabakhtin) [reply protected/humhub/components/ApplicationTrait.php:57]: https://github.com/humhub/humhub/pull/7243/commits/8a30aae0e95545692516daf5c5f52cb0851ca2d7#diff-e5be6221a42383b120c78c03fcf0111d24e56a67f21e988bed1f0b65f034cfbfR55

File: protected/humhub/libs/DynamicConfig.php [hunk 5]
```diff
@@ -91,18 +91,7 @@ public static function rewrite()
...
-        } else {
-            $config['language'] = Yii::$app->language;
-        }
-
-        $defaultTimeZone = Yii::$app->settings->get('defaultTimeZone');
-        if (!empty($defaultTimeZone)) {
-            $config['timeZone'] = $defaultTimeZone;
-            $config['components']['formatter']['defaultTimeZone'] = $defaultTimeZone;
-        }
+        $config['language'] = Yii::$app->settings->get('defaultLanguage', Yii::$app->language);
```
[the remark this answers is not in the document]
Comment (yurabakhtin) [reply protected/humhub/libs/DynamicConfig.php]: https://github.com/humhub/humhub/pull/7243/commits/8a30aae0e95545692516daf5c5f52cb0851ca2d7#diff-f6e347c8c1c2fa04e79d8ad66a9fcd956dade7f02ebe9640ade01aaca390786fL94 - I have removed it and it seems it works well.

> * There is a section with "unsets" for old dynamic content. Can you please add removed stuff here. So it's automatically removed on the next save.

https://github.com/humhub/humhub/pull/7243/commits/8a30aae0e95545692516daf5c5f52cb0851ca2d7#diff-864ee5d29be553a1a5776cf6cea6e7190eea1b0c7a03ef5edd005e8d2b87fe58R967-R972

File: protected/humhub/modules/installer/controllers/SetupController.php [hunk 6]
```diff
@@ -171,9 +172,22 @@ public function actionMigrate()
...
 
         $this->initDatabase();
-        return $this->redirect(['cron']);
+        return $this->redirect(['localisation']);
     }
 
+    /**
+     * Localisation
+     */
+    public function actionLocalisation()
```
[the remark this answers is not in the document]
Comment (yurabakhtin) [reply protected/humhub/modules/installer/controllers/SetupController.php]: https://github.com/humhub/humhub/pull/7243/commits/8a30aae0e95545692516daf5c5f52cb0851ca2d7#diff-3be885119f7c206d26a3b3b58c296f8a938da2cf7339fcaf91bfb404847bf146R137

File: protected/humhub/modules/installer/forms/LocalisationForm.php (hunk 2 above)
Comment (yurabakhtin) [reply protected/humhub/modules/installer/forms/LocalisationForm.php]: https://github.com/humhub/humhub/pull/7243/commits/8a30aae0e95545692516daf5c5f52cb0851ca2d7#diff-aa96ce8679a1100f0a92ba7e429ea682632777bb866fb499d5ab941446b68af3L66 - no, we don't need, I think it was copied by mistake, thanks.

File: protected/humhub/modules/user/models/User.php (hunk 3 above)
Comment (yurabakhtin) [reply protected/humhub/modules/user/models/User.php:281]: https://github.com/humhub/humhub/pull/7243/commits/8a30aae0e95545692516daf5c5f52cb0851ca2d7#diff-d3a4745ae7c92bb2ed3212a38c146f5f29268cf40a693d007f752f679a1f52abR282

File: protected/humhub/libs/SelfTest.php [hunk 7]
```diff
@@ -964,7 +964,12 @@ public static function getLegancyConfigSettings(): array
             'components' => [
                 'formatterApp' => new UnsetArrayValue(),
                 'search' => new UnsetArrayValue(),
+                'formatter' => [
+                    'defaultTimeZone' => new UnsetArrayValue(),
+                ],
             ],
+            'timeZone' => new UnsetArrayValue(),
```
Comment (luke-) [inline protected/humhub/libs/SelfTest.php:971]: @yurabakhtin No need to add the TZ stuff here.

Simply add here, then it's removed from existing DynamicConfigs as well: 
https://github.com/humhub/humhub/blob/master/protected/humhub/libs/DynamicConfig.php#L136-L142

**Review by luke- — CHANGES_REQUESTED** [review]
@yurabakhtin Thanks, looks good. One small point

File: protected/humhub/libs/SelfTest.php (hunk 7 above)
Comment (yurabakhtin) [reply protected/humhub/libs/SelfTest.php]: @luke- Commit https://github.com/humhub/humhub/pull/7243/commits/22dc26cc54f5103864bf25385aec6e117aa97b18.

**Review by luke- — APPROVED** [review]

**Pull request merged**